diff -Naur a/gcc/config/mips/t-pic32 b/gcc/config/mips/t-pic32
--- a/gcc/config/mips/t-pic32	1970-01-01 01:00:00.000000000 +0100
+++ b/gcc/config/mips/t-pic32	2021-01-10 22:24:16.941965160 +0000
@@ -0,0 +1,34 @@
+# Copyright (C) 1999-2020 Free Software Foundation, Inc.
+#
+# This file is part of GCC.
+#
+# GCC is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3, or (at your option)
+# any later version.
+#
+# GCC is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with GCC; see the file COPYING3.  If not see
+# <http://www.gnu.org/licenses/>.
+
+# Build the libraries for both hard and soft floating point
+
+MULTILIB_OPTIONS = mips32r2/mips32r5 msoft-float mdspr2 mips16
+MULTILIB_DIRNAMES = mx mz sf dsp mips16
+MULTILIB_MATCHES = 
+
+MULTILIB_REQUIRED =  mips32r2/msoft-float 
+MULTILIB_REQUIRED += mips32r2/msoft-float/mips16
+MULTILIB_REQUIRED += mips32r2/msoft-float/mdspr2 
+MULTILIB_REQUIRED += mips32r2/msoft-float/mdspr2/mips16
+MULTILIB_REQUIRED += mips32r2/mdspr2 
+MULTILIB_REQUIRED += mips32r5/mdspr2 
+MULTILIB_REQUIRED += mips32r5/msoft-float/mdspr2 
+MULTILIB_REQUIRED += mips32r5/msoft-float/mdspr2/mips16 
+#MULTILIB_REQUIRED += mmicromips/msoft-float 
+#MULTILIB_REQUIRED += mmicromips/msoft-float/mdspr2
diff -Naur a/gcc/config.gcc b/gcc/config.gcc
--- a/gcc/config.gcc	2020-07-23 07:35:17.316384243 +0100
+++ b/gcc/config.gcc	2021-01-10 14:35:21.189454379 +0000
@@ -2652,7 +2652,14 @@
 	;;
 mips-*-elf* | mipsel-*-elf* | mipsr5900-*-elf* | mipsr5900el-*-elf*)
 	tm_file="elfos.h newlib-stdint.h ${tm_file} mips/elf.h"
-	tmake_file="mips/t-elf"
+    case ${target} in
+        *-pic32-elf*)
+            tmake_file="mips/t-pic32"
+            ;;
+        *-*-elf*)
+            tmake_file="mips/t-elf"
+            ;;
+    esac
 	;;
 mips64r5900-*-elf* | mips64r5900el-*-elf*)
 	tm_file="elfos.h newlib-stdint.h ${tm_file} mips/elf.h mips/n32-elf.h"
